### PR TITLE
Simplify SQL queries of hidden_nids and coauthor_nids

### LINKED ISSUE #5443: Simplify raw SQL queries to reduce the response time of web pages
I came across a lot of raw and complicated SQL queries which can be simplified. The response time of the web pages is very slow. It takes about >10-15 secs to load a web page without cache. 
Also, a lot queries have been repeated which can be replaced by instance variables or helper functions. 
Let's work on simplifying queries and help improving the user experience.

### PR DESCRIPTION
**Pull request by gautamig54**
Fixes #5443  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by gautamig54** [comment]
@jywarren Kindly review my PR. I am not able to understand the codeclimate error, though it passes the travis-ci tests. Thanks!

**Comment by jainaman224** [comment]
@gautamig54 Basically even if you don't use `return` keyword. It will return that statement. As in rails function it returns the value of last evaluated statement by default.
Still, I feel it is better to use `return` keyword in open source to avoid confusion. @jywarren What are your thoughts?

File: app/models/node.rb [hunk 1]
```diff
@@ -117,6 +117,14 @@ def slug_from_path
     path.split('/').last
   end
 
+  def has_a_tag(name)
+    if tags.where(name: name).count.positive?
```
Comment (jainaman224) [inline app/models/node.rb]: This can be simplified as
> return tags.where(name: name).count.positive?

**Comment by gautamig54** [comment]
Thank you @jainaman224 for your review! I have made the changes suggested by you. 
@jywarren Kindly review the PR. Thanks!

**Comment by gautamig54** [comment]
@publiclab/reviewers Kindly review my PR. Thanks!

**Comment by jywarren** [comment]
Awesome, thank you!

**Pull request merged**